tests/components/template/test_blueprint.py

```diff
@@ -0,0 +1,235 @@
+                ]
+            },
+        )
+
+    hass.states.async_set("binary_sensor.foo", "off", {"friendly_name": "Foo"})
+    hass.states.async_set("binary_sensor.bar", "on", {"friendly_name": "Bar"})
+    await hass.async_block_till_done()
+
+    assert hass.states.get("binary_sensor.foo").state == "off"
+    assert hass.states.get("binary_sensor.bar").state == "on"
+
+    inverted_foo = hass.states.get("binary_sensor.inverted_foo")
+    assert inverted_foo
+    assert inverted_foo.state == "on"
+
+    inverted_bar = hass.states.get("binary_sensor.inverted_bar")
+    assert inverted_bar
+    assert inverted_bar.state == "off"
+
+    foo_template = template.blueprint_in_template(hass, "binary_sensor.foo")
+    inverted_foo_template = template.blueprint_in_template(
+        hass, "binary_sensor.inverted_foo"
+    )
+    assert foo_template is None
+    assert inverted_foo_template == "inverted_binary_sensor.yaml"
+
+    inverted_binary_sensor_blueprint_instances = template.templates_with_blueprint(
+        hass, "inverted_binary_sensor.yaml"
+    )
+    assert len(inverted_binary_sensor_blueprint_instances) == 2
+
+    assert len(template.templates_with_blueprint(hass, "dummy.yaml")) == 0
+
+    with pytest.raises(BlueprintInUse):
+        await template.async_get_blueprints(hass).async_remove_blueprint(
+            "inverted_binary_sensor.yaml"
+        )
+
+
+async def test_domain_blueprint(hass: HomeAssistant) -> None:
+    """Test DomainBlueprint services."""
+    reload_handler_calls = async_mock_service(hass, DOMAIN, SERVICE_RELOAD)
+    mock_create_file = AsyncMock()
+    mock_create_file.return_value = True
+
+    with patch(
+        "homeassistant.components.blueprint.models.DomainBlueprints._create_file",
+        mock_create_file,
+    ):
+        await template.async_get_blueprints(hass).async_add_blueprint(
+            Blueprint(
+                {
+                    "blueprint": {
+                        "domain": DOMAIN,
+                        "name": "Test",
+                    },
+                },
+                expected_domain="template",
+                path="xxx",
+                schema=BLUEPRINT_SCHEMA,
+            ),
+            "xxx",
+            True,
+        )
+    assert len(reload_handler_calls) == 1
+
+
+async def test_invalid_blueprint(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test an invalid blueprint definition."""
+
+    with patch_invalid_blueprint():
+        assert await async_setup_component(
+            hass,
+            "template",
+            {
+                "template": [
+                    {
+                        "use_blueprint": {
+                            "path": "invalid.yaml",
+                        },
+                        "name": "Invalid blueprint instance",
+                    },
+                ]
+            },
+        )
+
+    assert "more than one platform defined per blueprint" in caplog.text
```

Comment:
This error log is confusing since there's no platform at all defined, right?

Reply:
That above is the blueprint instance. The blueprint has 2 platforms defined https://github.com/tetele/ha-core/blob/2d678a9552908cdf7863d3af9e475ce4161c8d22/tests/components/template/test_blueprint.py#L68-L69

Reply:
Ok